$MirOS: ports/net/tor/patches/patch-src_or_test_c,v 1.1 2008/03/04 02:24:15 tg Exp $

	format_rfc1123_time() uses libc functions, while
	parse_rfc1123_time() uses a private function taken
	from Python and not supporting leap seconds, so this
	test written by <censored /> would otherwise fail

--- src/or/test.c.orig	Fri Nov 30 07:49:26 2007
+++ src/or/test.c	Sun Nov 20 04:51:54 2011
@@ -583,6 +583,14 @@ _compare_without_first_ch(const void *a,
   return strcasecmp(s1+1, s2);
 }
 
+#ifdef SYSKERN_MIRTIME_H
+#define OS2UTC(x)	timet2posix(x)
+#define UTC2OS(x)	posix2timet(x)
+#else
+#define OS2UTC(x)	(x)
+#define UTC2OS(x)	(x)
+#endif
+
 static void
 test_util(void)
 {
@@ -622,6 +630,7 @@ test_util(void)
 
   /* The test values here are confirmed to be correct on a platform
    * with a working timegm. */
+  /* Note from MirBSD developers: yeah, sure... *pats head* */
   a_time.tm_year = 2003-1900;
   a_time.tm_mon = 7;
   a_time.tm_mday = 30;
@@ -637,7 +646,7 @@ test_util(void)
 
   format_rfc1123_time(timestr, 0);
   test_streq("Thu, 01 Jan 1970 00:00:00 GMT", timestr);
-  format_rfc1123_time(timestr, (time_t)1091580502UL);
+  format_rfc1123_time(timestr, UTC2OS((time_t)1091580502UL));
   test_streq("Wed, 04 Aug 2004 00:48:22 GMT", timestr);
 
   t_res = 0;
